core/server/src/main/java/alluxio/master/AbstractMaster.java

```diff
@@ -174,16 +179,19 @@ public void stop() throws IOException {
       }
     }
     // Shut down the executor service, interrupting any running threads.
-    mExecutorService.shutdownNow();
-    String awaitFailureMessage =
-        "waiting for {} executor service to shut down. Daemons may still be running";
-    try {
-      if (!mExecutorService.awaitTermination(SHUTDOWN_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
-        LOG.warn("Timed out " + awaitFailureMessage, this.getClass().getSimpleName());
+    if (mExecutorService != null) {
+      mExecutorService.shutdownNow();
+      String awaitFailureMessage =
+          "waiting for {} executor service to shut down. Daemons may still be running";
+      try {
+        if (!mExecutorService.awaitTermination(SHUTDOWN_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+          LOG.warn("Timed out " + awaitFailureMessage, this.getClass().getSimpleName());
+        }
+      } catch (InterruptedException e) {
+        LOG.warn("Interrupted while " + awaitFailureMessage, this.getClass().getSimpleName());
       }
-    } catch (InterruptedException e) {
-      LOG.warn("Interrupted while " + awaitFailureMessage, this.getClass().getSimpleName());
     }
+    mExecutorService = null;
```

Comment:
Can anything above (lines 183 - 191) throw any other type of exception which might skip over this line that sets it to `null`?